splight_lib/client/datalake/remote_client.py

```diff
@@ -61,217 +67,116 @@ async def async_save(
-        url = self._base_url / f"{self._PREFIX}/dumpdata/"
-        params = self._parse_params(**filters)
-
-        response = self._restclient.get(url, params=params)
-        response.raise_for_status()
-        logger.debug(
-            "Retrieving dataframe from datalake.", tags=LogTags.DATALAKE
-        )
-
-        df: pd.DataFrame = pd.DataFrame(pd.read_csv(StringIO(response.text)))
-        if df.empty:
-            return df
-        df["timestamp"] = pd.to_datetime(df["timestamp"])
-        df.set_index("timestamp", inplace=True)
-        return df
-
-    @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    def save_dataframe(self, collection: str, dataframe: pd.DataFrame) -> None:
-        logger.debug("Saving dataframe.", tags=LogTags.DATALAKE)
-
-        # POST /datalake/loaddata/
-        url = self._base_url / f"{self._PREFIX}/loaddata/"
-
-        tmp_file = NamedTemporaryFile("w")
-        with open(tmp_file.name, "wb") as fid:
-            dataframe.to_csv(fid)
-        response = self._restclient.post(
-            url,
-            data={"source": collection},
-            files={"file": open(tmp_file.name, mode="rb")},
-        )
-        response.raise_for_status()
-
-    @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    def create_index(self, collection: str, indexes: List[Dict]) -> None:
-        # POST /datalake/index/
-        logger.debug(
-            "Creating index for collection: %s.",
-            collection,
-            tags=LogTags.DATALAKE,
-        )
-
-        url = self._base_url / f"{self._PREFIX}/index/"
-        data = {"source": collection, "index": indexes}
-        response = self._restclient.post(url, json=data)
-        response.raise_for_status()
-
-    @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    def raw_aggregate(
-        self, collection: str, pipeline: List[Dict]
-    ) -> List[Dict]:
-        logger.debug(
-            "Aggregate on datalake collection: %s.",
-            collection,
-            tags=LogTags.DATALAKE,
-        )
-        # POST /datalake/aggregate/?source=collection
-        url = self._base_url / f"{self._PREFIX}/aggregate/"
-        params = {"source": collection}
-        data = {"pipeline": pipeline}
-        response = self._restclient.post(url, params=params, data=data)
-        response.raise_for_status()
-        return response.json()
-
-    @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    def execute_query(
-        self,
-        from_timestamp: datetime,
-        to_timestamp: Optional[datetime],
-        query: Dict,
-    ) -> pd.DataFrame:
-        url = self._base_url / f"{self._PREFIX}/data/request/csv/"
-        to_timestamp = to_timestamp.isoformat() if to_timestamp else None
-        data = {
-            "from_timestamp": from_timestamp.isoformat(),
-            "to_timestamp": to_timestamp,
-            "traces": [
+        # GET /data/read
```

Comment:
```suggestion
        # POST /data/read
```

Reply:
No need for this, the endpoint is a POST

Reply:
yeah I know, I just wanted to say that the comment was wrong